test/conformance/enqueue/urEnqueueMemBufferMap.cpp

```diff
@@ -21,22 +61,30 @@ TEST_P(urEnqueueMemBufferMapTest, SuccessRead) {
     }
 }
 
+static std::vector<uur::mem_buffer_map_write_test_parameters_t>
+    map_write_test_parameters{
+        {8, UR_MEM_FLAG_READ_WRITE, UR_MAP_FLAG_WRITE},
+        {8, UR_MEM_FLAG_READ_WRITE, UR_MAP_FLAG_WRITE_INVALIDATE_REGION},
+    };
+
 using urEnqueueMemBufferMapTestWithWriteFlagParam =
-    uur::urMemBufferQueueTestWithParam<ur_map_flag_t>;
+    uur::urMemBufferQueueTestWithParam<
+        uur::mem_buffer_map_write_test_parameters_t>;
+
 UUR_TEST_SUITE_P(urEnqueueMemBufferMapTestWithWriteFlagParam,
-                 ::testing::Values(UR_MAP_FLAG_WRITE,
-                                   UR_MAP_FLAG_WRITE_INVALIDATE_REGION),
-                 uur::deviceTestWithParamPrinter<ur_map_flag_t>);
+                 ::testing::ValuesIn(map_write_test_parameters),
+                 uur::printMemBufferMapWriteTestString<
+                     urEnqueueMemBufferMapTestWithWriteFlagParam>);
 
 TEST_P(urEnqueueMemBufferMapTestWithWriteFlagParam, SuccessWrite) {
     const std::vector<uint32_t> input(count, 0);
     ASSERT_SUCCESS(urEnqueueMemBufferWrite(queue, buffer, true, 0, size,
                                            input.data(), 0, nullptr, nullptr));
 
     uint32_t *map = nullptr;
-    ASSERT_SUCCESS(urEnqueueMemBufferMap(queue, buffer, true, getParam(), 0,
-                                         size, 0, nullptr, nullptr,
-                                         (void **)&map));
+    ASSERT_SUCCESS(urEnqueueMemBufferMap(queue, buffer, true,
+                                         getParam().map_flag, 0, size, 0,
+                                         nullptr, nullptr, (void **)&map));
```

Comment:
@omarahmed1111 We both created a similar `urMemBufferQueueTestWithParam` fixture around the same time. The difference in mine was I was accessing parameters in the fixture to facilitate creating different buffer types. I've attempted to merge them together but not sure it's a great solution, particularly from adding a `ur_mem_flag_t` that you needed in this test, but I didn't need it for the tests I modified although probably not a big deal given it's just a single enum. Does this seem okay or would it be better to separate the fixtures?

Reply:
Merging them does make sense to me, the test that I added only needed to inherit from the parameter fixture chain and initialize a buffer, so any fixture that does these 2 things would work, so it doesn't need to be a special one for it.

Reply:
Thanks for this. I guess I was thinking more about having to change your test so the template is based off the new struct I added `mem_buffer_map_write_test_parameters_t` instead of just `ur_map_flag_t` because in the fixture I'm setting the buffer size and flags from parameters and this wasn't needed for yours. But if you think it looks okay I'll just go ahead with that change, thanks!